backend/src/main/java/de/neuefische/booklibrary/backend/api/ApiBookService.java

```diff
@@ -0,0 +1,107 @@
+package de.neuefische.booklibrary.backend.api;
+
+
+import de.neuefische.booklibrary.backend.book.Availability;
+import de.neuefische.booklibrary.backend.book.Book;
+import de.neuefische.booklibrary.backend.book.RentBookInfo;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Service
+public class ApiBookService {
+    private static final String PATH_QUERY = "?q=";
+    private static final String PATH_KEY = "&key=";
+    public static final String DEFAULT_COVER = "https://simg.nicepng.com/png/small/251-2515797_no-circle-book-book-logo-png.png";
+    private final String apiKey;
+    private final WebClient webClient;
+
+    public ApiBookService(@Value("${api.url}") String apiUrl, @Value("${api.key}") String apiKey) {
+        this.apiKey = apiKey;
+        this.webClient = WebClient.create(apiUrl);
+    }
+
+    public List<Book> getApiBookByIsbn(String isbn) throws ResponseStatusException {
+        String searchByIsbn = "isbn:";
+        String query = PATH_QUERY + searchByIsbn + isbn + PATH_KEY + apiKey;
+        ResponseEntity<BookResponseElement> bookResponse = getBookResponse(query);
+        List<ApiBook> books = getBookList(bookResponse).stream().filter(book -> book.volumeInfo()
+                .industryIdentifiers()
+                .stream()
+                .anyMatch(isbn1 -> isbn.equals(isbn1.identifier()))).toList();
+        List<Book> bookList = new ArrayList<>();
+        getBookListFromApiBookList(books, bookList);
+        return bookList;
+
+    }
+
+    private static void getBookListFromApiBookList(List<ApiBook> books, List<Book> bookList) {
```

Comment:
Die Namensgebung könnte hier noch deutlicher machen, worum es sich bei `books` bzw. `bookList` handelt.
```suggestion
    private static void getBookListFromApiBookList(List<ApiBook> apiBooks, List<Book> applicationBooks) {
```

Reply:
Wichtig: Das Ergebnis einer Methode soll immer der Return-Wert sein. Parameter (inputs) sollen in einer Methode nicht bearbeitet / verändert werden.
Das Bearbeiten von Inputs nennt man auch "side-effects" und das macht den Code in der Regel schlechter verständlich.
Ausnahmen gibt es z.B. bei extremen Performance-Anforderungen.
Übrigens: Immutables verhindern diese Side-Effects grundsätzlich, einer der Gründe, warum ich Records so gerne mag.

Reply:
Das ist eine gute Bemerkung. Ich habe eine Rückgabewert von void auf List<Book> geändert, weil das dann entspricht dem was die Methode zurück liefert, nämlich List of Book